### PR TITLE
Properly translate users into runc format for exec

### LINKED ISSUE #1314: Using `podman exec --user=root` fails with invalid syntax error
/kind bug

**Description**

Running a command in a running container fails when specifying a user name but works with UID.

**Steps to reproduce the issue:**

1. Deploy any container with podman.

2. Run: `podman exec my_container ls`. It works.

3. Run: `podman exec --user=0 my_container ls`. It works.

4. Run: `podman exec --user=root my_container ls`. It fails.

**Describe the results you received:**

> exec failed: parsing root as int for uid failed: strconv.Atoi: parsing "root": invalid syntax
> exit status 1

**Describe the results you expected:**

> anaconda-post.log
> bin
> boot
> dev
> etc
> ...

**Output of `podman version`:**

```
Version:       0.8.2.1
Go Version:    go1.10.2
OS/Arch:       linux/amd64
```

**Output of `podman info`:**

```
host:
  Conmon:
    package: podman-0.8.2.1-1.gitf38eb4f.el7.x86_64
    path: /usr/libexec/podman/conmon
    version: 'conmon version 1.12.0-dev, commit: 74bba30adae4fdcacf887d35e4d6331318765028-dirty'
  MemFree: 230641664
  MemTotal: 8201699328
  OCIRuntime:
    package: runc-1.0.0-27.rc5.dev.git4bb1fe4.el7.x86_64
    path: /usr/bin/runc
    version: 'runc version spec: 1.0.0'
  SwapFree: 1610608640
  SwapTotal: 1610608640
  arch: amd64
  cpus: 4
  hostname: undercloud75.localdomain
  kernel: 3.10.0-862.11.6.el7.x86_64
  os: linux
  uptime: 1h 26m 5.42s (Approximately 0.04 days)
insecure registries:
  registries: []
registries:
  registries:
  - docker.io
  - registry.centos.org
  - registry.access.redhat.com
store:
  ContainerStore:
    number: 41
  GraphDriverName: overlay
  GraphOptions:
  - overlay.override_kernel_check=true
  GraphRoot: /var/lib/containers/storage
  GraphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
  ImageStore:
    number: 28
  RunRoot: /var/run/containers/storage
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Running it in local vagrant/virtualbox.

### PR DESCRIPTION
**Pull request by mheon**
Runc exec expects the --user flag to be formatted as UID:GID.
Use chrootuser code to translate whatever user is passed to exec
into this format.

Fixes #1314

### CODE REVIEW

**Comment by mheon** [comment]
Needs tests still - will write them after lunch

**Comment by EmilienM** [comment]
I tested your code, it works fine for me:

```
podman exec --user=root keystone ls
anaconda-post.log
bin
boot
dev
etc
home
lib
lib64
media
mnt
openstack
opt
proc
root
run
run_command
sbin
srv
sys
tmp
usr
var
```

File: libpod/container_api.go [hunk 1]
```diff
@@ -298,6 +300,19 @@ func (c *Container) Exec(tty, privileged bool, env, cmd []string, user string) e
 		capList = caps.GetAllCapabilities()
 	}
 
+	// If user was set, look it up in the container to get a UID to use on
+	// the host
+	hostUser := ""
+	if user != "" {
+		uid, gid, err := chrootuser.GetUser(c.state.Mountpoint, user)
+		if err != nil {
+			return errors.Wrapf(err, "error getting user to launch exec session aas")
```
Comment (TomSweeneyRedHat) [inline libpod/container_api.go]: I'm not sure what the "aas" at the end is for here.

**Comment by TomSweeneyRedHat** [comment]
Travis timed out, not sure why...

**Comment by mheon** [comment]
Added tests. I think the Fedora registry is having issues, though, so CI is probably broken.

**Comment by TomSweeneyRedHat** [comment]
bot, retest this please

**Comment by vrothberg** [comment]
bot, retest this please

**Comment by mheon** [comment]
One of the tests I added is failing - will look into it

**Comment by mheon** [comment]
Alright, this is ready
@TomSweeneyRedHat @rhatdan @vrothberg PTAL

**Comment by vrothberg** [comment]
LGTM

**Comment by vrothberg** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**